version: '3'
services:
  namenode:
    image: bde2020/hadoop-namenode:2.0.0-hadoop2.7.4-java8
    networks:
      - spark-net
    ports:
      - "50070:50070"
    volumes:
      - /change/to/your/path:/hadoop/dfs/name #change to your own path
    environment:
      - CLUSTER_NAME=test
    env_file:
      - ./hadoop.env

    deploy:
      mode: replicated
      replicas: 1
      restart_policy:
        condition: on-failure
      placement:
        constraints:
          - node.labels.worker == false
      labels:
        traefik.enable: "true"
        traefik.docker.network: spark-net 
        traefik.port: 50070
        traefik.frontend.rule: Path:/namenode/
        traefik.backend: namenode


  datanode:
    image: bde2020/hadoop-datanode:2.0.0-hadoop2.7.4-java8
    networks:
      - spark-net
    volumes:
      - /change/to/your/path:/hadoop/dfs/data #change to your own path
    env_file:
      - ./hadoop.env
    environment:
      SERVICE_PRECONDITION: "namenode:50070"
    depends_on:
      - namenode
    deploy:
      replicas: 3
      mode: replicated
      placement:
        constraints:
          - node.labels.worker == true
      restart_policy:
        condition: on-failure
      labels:
        traefik.docker.network: spark-net
        traefik.port: 50075


networks:
  spark-net:
    external: true
